src/cursor/abstract_cursor.ts

```diff
@@ -594,7 +601,7 @@ function nextDocument(cursor: AbstractCursor): Document | null | undefined {
   if (doc) {
     const transform = cursor[kTransform];
     if (transform) {
-      return transform(doc);
+      return transform(doc) as TSchema; // TODO: How can user's parameterize this
```

Comment:
Upon further investigation I don't think there's a problem here. But another set of eyes could help confirm that. It looks like there's no way to save a transform on to a cursor. I thought it was an option during construction that would effect the type emitted from each `.next()` but I don't see how you can specify one. So this is dead code?

Reply:
Let's create a ticket to investigate what, if anything, can we done to provide typing on this due to the complication with chaining

Reply:
Ticket: NODE-3283
And I've permitted overriding the generic in `forEach`, `next`, `tryNext`, and `toArray`. 
```
db.collection<Pet>().find().map((p) => p.age).toArray() // Still returns Pet[] in TS but number[] at runtime
db.collection<Pet>().find().map((p) => p.age).toArray<number>() // correct TS, number[]
```
Unfortunately it's not very useful for `[Symbol.asyncIterator]()` since that isn't called by users.